src/Symfony/Component/DependencyInjection/Secrets/JweHandler.php

```diff
@@ -0,0 +1,286 @@
+    }
+
+    /**
+     * @return string
+     */
+    private function getPublicKey()
+    {
+        return $this->readKey($this->publicKeyLocation);
+    }
+
+    /**
+     * @throws InvalidArgumentException if the fileLocation does not exist in a writable directory or
+     * if the file already exists and $overwriteExisting is set to false
+     */
+    public function initSecretsFiles(bool $overwriteExisting)
+    {
+        if (!$overwriteExisting) {
+            if (file_exists($this->secretsLocation)) {
+                throw new InvalidArgumentException(sprintf(
+                    'secrets file at %s already exists',
+                    $this->secretsLocation
+                ));
+            }
+
+            if (file_exists($this->publicKeyLocation)) {
+                throw new InvalidArgumentException(sprintf(
+                    'public key file at %s already exists',
+                    $this->publicKeyLocation
+                ));
+            }
+
+            if (file_exists($this->privateKeyLocation)) {
+                throw new InvalidArgumentException(sprintf(
+                    'private key file at %s already exists',
+                    $this->privateKeyLocation
+                ));
+            }
+        }
+
+        $this->writeFile($this->secretsLocation, json_encode([], self::JSON_ENCODE_OPTIONS));
+        $this->writeNewKeyPair();
+        return $this;
+    }
+
+    /**
+     * @return bool
+     */
+    private function isValidKeyLocation(string $filePath)
+    {
+        return $this->isValidSecretsFile($filePath) &&
+            (SODIUM_CRYPTO_BOX_SECRETKEYBYTES === filesize($filePath));
+    }
+
+    /**
+     * @return bool
+     */
+    private function isValidSecretsFile(string $filePath)
+    {
+        return (stream_is_local($filePath) && is_readable($filePath));
+    }
+
+    /**
+     * @return array
+     */
+    private function populateSecrets()
+    {
+        $this->secrets = $this->readSecrets($this->secretsLocation);
+        return $this;
+    }
+
+    private function readKey(string $keyLocation)
+    {
+        return trim(file_get_contents($keyLocation));
+    }
+
+    /**
+     * returns the decoded json secrets from $secretsLocation or false if the file does not contain valid json
+     * @return array|false
+     */
+    private function readSecrets(string $secretsLocation)
+    {
+        $rawSecrets = trim(file_get_contents($secretsLocation));
+
+        return json_decode($rawSecrets, true);
+    }
+
+    /**
+     * @return $this
+     */
+    private function writeEncrypted()
+    {
+        $this->writeFile($this->secretsLocation, json_encode($this->secrets, self::JSON_ENCODE_OPTIONS));
+        return $this;
+    }
+
+    private function writeFile(string $fileLocation, string $content)
+    {
+        $fileDir = pathinfo($fileLocation, PATHINFO_DIRNAME);
+        if (!is_dir($fileDir) || !is_writeable($fileLocation)) {
```

Comment:
good call, splitting these out for better error feedback. I am actually adding logic to create a dir if missing and within the local project config directory. Since defaults are in `config/secrets/` right now which won't exist, and there shouldn't be permissions issues for those. Then passing a flag from the `secrets:init` command call to `initSecretsFiles` to do so. Let me know if that seems helpful or problematic to you, my plan was to expect developers to create their own directories (which would more likely be outside of the project) for the private keys but remove the need for them to create `config/secrets/` directory here